src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlObjectSerializer.cs

```diff
@@ -355,7 +355,7 @@ internal bool IsRootElement(XmlReaderDelegator reader, DataContract contract, Xm
             reader.MoveToElement();
             if (name != null) // root name set explicitly
             {
-                return reader.IsStartElement(name, ns!); // https://github.com/dotnet/runtime/issues/41395
+                return reader.IsStartElement(name, ns ?? XmlDictionaryString.Empty);
             }
```

Comment:
Is it worth modifying the null check on name to also check that it isn't equivalent to l to XmlDictionaryString.Empty? If you do that, it might be worth checking `name.Value.Length != 0` as someone can make their own XmlDictionaryString with the empty string and then the equality comparison won't work with XmlDictionaryString.Empty

Reply:
Interestingly, we don't check the rootname or rootnamespace arguments for validity when constructing DCS. So somebody could totally create an empty rootname. But I think that's an error that should be dealt with in user code, even if we could validate a little better. I don't think this is the place or the issue to address that pit of success.